api/src/main/java/WillYingling/rogue/RobotComponent.java

```diff
@@ -0,0 +1,16 @@
+public abstract class RobotComponent {
```

Comment:
Ok I understand that for replacing the init and halt methods, but what about update? Or would each component run it's own loop?